src/Npgsql/VolatileResourceManager.cs

```diff
@@ -95,6 +96,8 @@ public void Prepare(PreparingEnlistment preparingEnlistment)
         }
     }
 
+    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Changing Enlist to be false does not affect potentially trimmed out functionality.")]
```

Comment:
Is this actually safe? I mean, if to commit a 2PC transaction we potentially need to call CloneWith which isn't safe, 2PC isn't safe, no?

But in any case, if we go ahead with the plan to make the full data source AOT-compatible (#5289), CloneWith should be fine and this as well, so it's moot.

Reply:
As discussed offline, in theory it's safe due to it being a narrowing clone. We're not changing the given datasource to potentially include trimmed out functionality.